sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -700,7 +901,13 @@ class RocksDB(
     new NextIterator[ByteArrayPair] {
       override protected def getNext(): ByteArrayPair = {
         if (iter.isValid) {
-          byteArrayPair.set(iter.key, iter.value)
+          val key = if (useColumnFamilies) {
+            decodeStateRowWithPrefix(iter.key)._1
```

Comment:
This concerns me a lot, as not giving the column family ID/name back to the caller makes the outcome be useless for multiple column family case. Say, column family cfA and cfB somehow have the same key keyA. This doesn't even give any meaning to global scan.

That's why I commented like the above.

Reply:
have added the new iterator API now. so I guess we should be good here ?